filament/backend/src/Platform.cpp

```diff
@@ -139,46 +139,55 @@ bool Platform::queryFrameTimestamps(SwapChain const*, uint64_t, FrameTimestamps*
 }
 
 void Platform::setBlobFunc(InsertBlobFunc&& insertBlob, RetrieveBlobFunc&& retrieveBlob) noexcept {
+    std::lock_guard<std::mutex> lock(mBlobFuncsMutex);
     mInsertBlob = std::move(insertBlob);
     mRetrieveBlob = std::move(retrieveBlob);
 }
 
 bool Platform::hasInsertBlobFunc() const noexcept {
+    std::lock_guard<decltype(mBlobFuncsMutex)> lock(mBlobFuncsMutex);
     return bool(mInsertBlob);
 }
 
 bool Platform::hasRetrieveBlobFunc() const noexcept {
+    std::lock_guard<decltype(mBlobFuncsMutex)> lock(mBlobFuncsMutex);
     return bool(mRetrieveBlob);
 }
 
 void Platform::insertBlob(void const* key, size_t keySize, void const* value, size_t valueSize) {
+    std::lock_guard<decltype(mBlobFuncsMutex)> lock(mBlobFuncsMutex);
```

Comment:
we can't call `mInsertBlob` with the lock held. We need to get the callback to a local with the lock held, release the lock and then call the callback.

Reply:
> We need to get the callback to a local

Since `utils::Invocable` isn't copyable, the only way to do so is by wrapping the callbacks in a shared_ptr